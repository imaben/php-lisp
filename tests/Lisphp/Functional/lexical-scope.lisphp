"Lexical closure -- define, setf!, let, lambda test."

(define (adder n)
        {lambda [x]
                (setf! n (+ n x))
                n})

(define a (adder 3))

(echo (a 1) "\n")
(echo (a 2) "\n")


(define z 1)

{let [(z 123)]
     (echo z "\n")
     (setf! z 1234)
     (echo z "\n")}

(echo z "\n")


{let [(y 12)]
     (define y 123)
     (echo y "\n")}

(echo y "\n")
