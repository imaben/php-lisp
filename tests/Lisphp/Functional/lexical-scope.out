4
6
123
1234
1
12
123
